share/new_eds/nd.go

```diff
@@ -0,0 +1,30 @@
+package eds
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/shwap"
+)
+
+// NamespacedData extracts shares for a specific namespace from an EDS, considering
+// each row independently.
+func NamespacedData(
+	ctx context.Context,
+	root *share.Root,
```

Comment:
prolly a silly question, but the comment doc suggests that I can call this with JUST and EDS, but it'd be good to explain (really NOT assume) that I know why I need, or even WHAT the share.Root is for and why it's necessary

Reply:
doc improved
https://github.com/celestiaorg/celestia-node/pull/3452/commits/8cbd6b461597a0b4ea0a6e8f0e500f78018d6309